src/server/game/Battlegrounds/ArenaTeamMgr.cpp

```diff
@@ -186,8 +186,9 @@ void ArenaTeamMgr::DistributeArenaPoints()
     {
         if (ArenaTeam* at = titr->second)
         {
-            at->FinishWeek();
-            at->SaveToDB();
+            if (at->FinishWeek())
+                at->SaveToDB();
+            
             at->NotifyStatsChanged();
```

Comment:
I think its fine as the function will only let them know that the flush has been completed, even if their score is intact.